File: backend/app/schemas/progress.py
"""Progress update request/response schemas."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field

from app.models.task import TaskAiStatus


class ProgressSubmit(BaseModel):
    content: str = Field(min_length=1, max_length=5000)
    mark_completed: bool = False


class ProgressUpdateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    user_id: int
    raw_content: str
    summary: str | None
    progress_percent: int | None
    ai_status: TaskAiStatus | None
    risk_detected: bool | None
    ai_analysis_failed: bool
    created_at: datetime


class RecentProgressItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    task_id: int
    task_name: str
    user_id: int
    user_name: str
    raw_content: str
    summary: str | None
    created_at: datetime

File: frontend/types/progress.ts
export interface ProgressUpdate {
  id: number;
  task_id: number;
  user_id: number;
  raw_content: string;
  summary: string | null;
  progress_percent: number | null;
  ai_status: string | null;
  risk_detected: boolean | null;
  ai_analysis_failed: boolean;
  created_at: string;
}

export interface ProgressSubmitInput {
  content: string;
  mark_completed?: boolean;
}

export interface RecentProgressItem {
  id: number;
  task_id: number;
  task_name: string;
  user_id: number;
  user_name: string;
  raw_content: string;
  summary: string | null;
  created_at: string;
}
